### PR TITLE
grass.py: used space as delimiter to get GRASS_VERSION_STRING

### PR DESCRIPTION
**Pull request by ninsbl**
grass --config currently throws an error:

```
Traceback (most recent call last):
  File "/usr/local/Cellar/osgeo-grass/7.8.2_3/libexec/bin/grass78", line 2025, in main
    index = sys.argv.index(batch_exec_param)
ValueError: '--exec' is not in list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/osgeo-grass/7.8.2_3/libexec/bin/grass78", line 2216, in <module>
    main()
  File "/usr/local/Cellar/osgeo-grass/7.8.2_3/libexec/bin/grass78", line 2030, in main
    params = parse_cmdline(sys.argv[1:], default_gui=default_gui)
  File "/usr/local/Cellar/osgeo-grass/7.8.2_3/libexec/bin/grass78", line 1951, in parse_cmdline
    print_params()
  File "/usr/local/Cellar/osgeo-grass/7.8.2_3/libexec/bin/grass78", line 1862, in print_params
    "%s\n" % val[0].split(':')[1].rstrip('$"\n').strip())
IndexError: list index out of range
```
This is very inconvenient if you want to determine where GRASS is installed e.g. for usage in R or the like.

The reason for the error is that colon is used as delimiter to parse GIS_H_VERSION and GRASS_VERSION_STRING.

Unfortunately, GRASS_VERSION_STRING is not replaced with the proper value and I have no idea ho to do that...

### CODE REVIEW

**Comment by ninsbl** [comment]
BTW: applying #325 does not provide revision info in:
`grass --config`

**Comment by metzm** [comment]
Looks good to me, but it seems that this PR requires PR #325. Therefore it can only be merged after #325 has been merged, correct?

**Comment by ninsbl** [comment]
This has been independent from #325 until now.
With the latest commit however, the `GRASS_HEADERS_VERSION` is taken from version.h by `grass --config` and (new) also `GRASS_HEADERS_DATE` (so #325 needs to be merged first now).

The `GRASS_VERSION_STRING` in version.h is currently (with #325 applied) `@(#) 7.9.dev (2020)`, while GRASS_HEADERS_VERSION is `7.9.dev (2020) fdae3b2441`. Note sure if `GRASS_VERSION_STRING`should be reported instead of `GRASS_HEADERS_DATE`. It seems though the former should be modified in #325...

**Comment by metzm** [comment]
> This has been independent from #325 until now.

I see, because git is not expanding the sv keyword Revision. 

> With the latest commit however, the `GRASS_HEADERS_VERSION` is taken from version.h by `grass --config` and (new) also `GRASS_HEADERS_DATE` (so #325 needs to be merged first now).

I don't think these are the intended values for version and date, instead grass.py probably wants to report the latest git hash of the whole source tree, not only the headers. Instead of using GRASS_HEADRS_VERSION from version.h, we could insert GRASS_VERSION_GIT from Platform.make when creating grass.py, i.e. modify lib/init/Makefile at the lines following L68:
https://github.com/OSGeo/grass/blob/master/lib/init/Makefile#L68

**Comment by ninsbl** [comment]
You are right, of course. Now I tried to address that in the two new commits.

**Comment by ninsbl** [comment]
Sorry, @wenzeslaus, I saw #323 just now, which obviously adresses the same issue, though solves it slightly different...
Feel free to disregard (close) this one…
But it would be great to get either this or #323 into 7.8.3...

**Comment by metzm** [comment]
@ninsbl @wenzeslaus The question is what exactly `grass7x --config revision` should report. The git hash of the latest commit to this branch, or the git hash of the latest commit to `/include` of this branch?

For `grass7x --config revision`, I prefer the git hash of the latest commit to this branch, not only to `/include` of this branch.

File: lib/init/Makefile [hunk 1]
```diff
@@ -68,6 +68,7 @@ endif
 	sed \
 	-e 's#@GISBASE@#$(RUN_GISBASE)#' \
 	-e 's#@GRASS_VERSION_NUMBER@#$(GRASS_VERSION_NUMBER)#' \
+	-e 's#@GRASS_VERSION_GIT@#$(GRASS_VERSION_GIT)#' \
```
Comment (metzm) [inline lib/init/Makefile]: This also needs to be  inserted below at L81/82

**Comment by ninsbl** [comment]
Should fix:
https://trac.osgeo.org/grass/ticket/4001

**Comment by wenzeslaus** [comment]
@ninsbl Feel free to use pieces of #323 if they seem useful. Maybe there are two issues to solve: 1) robust parsing (to e.g. gracefully fail, try harder to parse, etc., kind of what I tried in #323) and 2) Actually parse the right/current value (e.g. after #325).

I second @metzm in saying that we need to clarify what this should actually report. Perhaps it is a topic for different PR/issue. See, for example:

```
$ # 7.6.1 installed from PPA
$ grass76 --config version
7.6.1
$ grass76 --config revision
72327
$ grass76 --config svn_revision
xported

$ # 7.9 from #323 
$ grass79 --config version
7.9.dev
$ grass79 --config revision
7.9.dev (2020)
$ grass79 --config svn_revision
2f4104
```

**Comment by neteler** [comment]
I tested this PR (since PR #325 has been merged), and this is the result

```
grass79 --config
x86_64-pc-linux-gnu
./configure  --with-cxx --enable-largefile --with-proj --with-proj-share=/usr/share/proj --with-gdal=/usr/bin/gdal-config --with-python --with-geos --with-sqlite --with-nls --with-zstd --with-pdal --with-cairo --with-cairo-ldflags=-lfontconfig --with-freetype --with-freetype-includes=/usr/include/freetype2 --with-wxwidgets --with-fftw --with-postgres --with-postgres-includes=/usr/include/pgsql --with-blas --with-blas-includes=/usr/include/atlas-x86_64-base/ --with-lapack --with-lapack-includes=/usr/include/atlas-x86_64-base/ --with-netcdf --without-motif --without-bzlib --without-mysql --without-odbc --without-openmp --without-ffmpeg
gcc
/home/mneteler/software/grass_master/dist.x86_64-pc-linux-gnu
f062bffc8
7.9.dev
```

... no more error.

And here with the available parameters:

```
for i in arch build compiler path revision svn_revision version date ; do 
   echo "" ; echo "Testing --config $i:" ; grass79 --config $i
done

Testing --config arch:
x86_64-pc-linux-gnu

Testing --config build:
./configure  --with-cxx --enable-largefile --with-proj --with-proj-share=/usr/share/proj --with-gdal=/usr/bin/gdal-config --with-python --with-geos --with-sqlite --with-nls --with-zstd --with-pdal --with-cairo --with-cairo-ldflags=-lfontconfig --with-freetype --with-freetype-includes=/usr/include/freetype2 --with-wxwidgets --with-fftw --with-postgres --with-postgres-includes=/usr/include/pgsql --with-blas --with-blas-includes=/usr/include/atlas-x86_64-base/ --with-lapack --with-lapack-includes=/usr/include/atlas-x86_64-base/ --with-netcdf --without-motif --without-bzlib --without-mysql --without-odbc --without-openmp --without-ffmpeg

Testing --config compiler:
gcc

Testing --config path:
/home/mneteler/software/grass_master/dist.x86_64-pc-linux-gnu

Testing --config revision:
f062bffc8

Testing --config svn_revision:
062bffc8

Testing --config version:
7.9.dev

Testing --config date:
Parameter <date> not supported
```

Maybe `-config date` needs to be aligned to the merged PR #325? (I didn't check the code)

**Comment by ninsbl** [comment]
OK, will update the PR to cover changes in #325, hopefully tonight. Would be nice to get this into 7.8.3...

**Comment by neteler** [comment]
@ninsbl shall this be included in 7.8.3 yet?

**Comment by landam** [comment]
```
./bin.x86_64-pc-linux-gnu/grass79 --config date
    "2020-03-31T20:34:57+02:00"
```

**Comment by landam** [comment]
> The format of a `date` could be probably beatified (no indent, no quotes)
> 
> ```
> ./bin.x86_64-pc-linux-gnu/grass79 --config date
>     "2020-03-31T20:34:57+02:00"
> ```

Improved in 317631e52f363dffd10f887c780e185671efccc8

**Comment by neteler** [comment]
I tested this PR again with the available parameters:

```
for i in arch build compiler path revision svn_revision version date ; do 
   echo "" ; echo "Testing --config $i:" ; grass79 --config $i
done

Testing --config arch:
x86_64-pc-linux-gnu

Testing --config build:
./configure  --with-cxx --enable-largefile --with-proj --with-proj-share=/usr/share/proj --with-gdal=/usr/bin/gdal-config --with-python --with-geos --with-sqlite --with-nls --with-zstd --with-pdal --with-cairo --with-cairo-ldflags=-lfontconfig --with-freetype --with-freetype-includes=/usr/include/freetype2 --with-wxwidgets --with-fftw --with-postgres --with-postgres-includes=/usr/include/pgsql --with-blas --with-blas-includes=/usr/include/atlas-x86_64-base/ --with-lapack --with-lapack-includes=/usr/include/atlas-x86_64-base/ --with-netcdf --without-motif --without-bzlib --without-mysql --without-odbc --without-openmp --without-ffmpeg

Testing --config compiler:
gcc

Testing --config path:
/home/mneteler/software/grass_master/dist.x86_64-pc-linux-gnu

Testing --config revision:
745ee7ec9

Testing --config svn_revision:
062bffc8

Testing --config version:
7.9.dev

Testing --config date:
Tue Mar 31 20:34:57 2020 +0200
```

Looks now all good to me.

File: lib/init/grass.py [hunk 2]
```diff
@@ -339,7 +339,7 @@ def help_message(default_gui):
             gui=_("use $DEFAULT_GUI graphical user interface"),
             gui_detail=_("and set as default"),
             config=_("print GRASS configuration parameters"),
-            config_detail=_("options: arch,build,compiler,path,revision,svn_revision,version"),
+            config_detail=_("options: arch,build,compiler,path,revision,svn_revision,version,date"),
```
Comment (neteler) [inline lib/init/grass.py:342]:
```suggestion
            config_detail=_("options: arch,build,compiler,date,path,revision,svn_revision,version"),
```

Alpha-sort options

**Comment by hellik** [comment]
> 
> Testing --config revision:
> 745ee7ec9
> 
> Testing --config svn_revision:
> 062bffc8
> 
> Looks now all good to me.

should _svn_revision_ be renamed to be something more git related as svn isn't used anymore?

**Comment by landam** [comment]
> should _svn_revision_ be renamed to be something more git related as svn isn't used anymore?

yes, eg. `libgis_revision` (?) But keep `svn_revision` for backward compatibility. Let's create a new PR for that.

**Comment by ninsbl** [comment]
Backported in ea2dd9e3c399caa2e9760a53b32217aa16a7e0c6
I hope I did not make a mess when backporting...

**Comment by landam** [comment]
No mess, thanks for PR!

**Pull request merged**